### PR TITLE
added Urdu language config

### PR DESCRIPTION
**Pull request by thisalihassan**
## Description

Adds Urdu language setup to config.json to enable Urdu translations on Crowdin

### Check List

- [x] I have read the [Contributing Guidelines](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md) and made commit messages that follow the guideline.
- [x] I have run `npx turbo format` to ensure the code follows the style guide.
- [x] I have run `npx turbo test` to check if all tests are passing.
- [x] I have run `npx turbo build` to check if the website builds without errors.
- [x] I've covered new added functionality with unit tests if necessary.

### CODE REVIEW

**Review by devlopersabbir — COMMENTED** [review]
that's great bro :)

**Review by aymen94 — APPROVED** [review]

**Comment by thisalihassan** [comment]
Hi, can I have a follow up on this?

**Comment by ovflowd** [comment]
> Hi, can I have a follow up on this?

We are not merging any related i18n PR until we get our Crowdin integration fixed. Sorry for the waiting.

**Comment by bmuenzenmeyer** [comment]
we need more information here

image

I suspect we might need to also update https://github.com/nodejs/nodejs.org/blob/main/crowdin.yml#L15-L20

**Comment by bmuenzenmeyer** [comment]
This PR will need to be rebased or recreated now that https://github.com/nodejs/nodejs.org/pull/6850 merged.

**Comment by thisalihassan** [comment]
> we need more information here
> 
> image
> 
> I suspect we might need to also update https://github.com/nodejs/nodejs.org/blob/main/crowdin.yml#L15-L20

"ur-IN" is not officially recognized in the same way as "ur-PK" in many internationalization standards.
Urdu as spoken in Pakistan and India is fundamentally the same in terms of syntax and grammar

cc: @bmuenzenmeyer

**Comment by bmuenzenmeyer** [comment]
`ur-PK` added in Crowdin

File: apps/site/i18n/config.json [hunk 1]
```diff
@@ -259,6 +259,26 @@
...
+  {
+    "code": "ur",
+    "localName": "اردو",
+    "name": "Urdu",
+    "langDir": "rtl",
+    "dateFormat": "DD/MM/YYYY",
+    "hrefLang": "ur",
+    "enabled": false,
+    "default": false
+  },
```
Comment (bmuenzenmeyer) [inline apps/site/i18n/config.json:271]: Since crowdin does not recognize this code - I think we should remove it for now

Comment (thisalihassan) [reply apps/site/i18n/config.json:271]: ur or ur-PK?

Comment (bmuenzenmeyer) [reply apps/site/i18n/config.json:271]: it doesn't recognize `ur`. only `ur-PK` and `ur-IN`

Perhaps an alternative is to map `ur` to `ur-PK` like we do within `two_letters_code` throughout https://github.com/nodejs/nodejs.org/blob/main/crowdin.yml

there might be a way for me to do this in crowdin too. https://developer.crowdin.com/configuration-file/#language-mapping

sorry for the confusion- I am learning the workings of their interface

**Comment by Andrulko** [comment]
`ur` code is not officially supported by Crowdin, so it was added as a [custom language](https://support.crowdin.com/changing-target-languages/#adding-custom-languages) to the project:
https://crowdin.com/project/nodejs-web/ur

Now you can see expected code I hope!

File: apps/site/i18n/config.json [hunk 2]
```diff
@@ -259,6 +259,26 @@
...
+  {
+    "code": "ur-PK",
+    "localName": "اردو (پاکستان)",
+    "name": "Urdu (Pakistan)",
+    "langDir": "rtl",
+    "dateFormat": "DD/MM/YYYY",
+    "hrefLang": "ur-PK",
+    "enabled": false,
+    "default": false
+  },
```
Comment (bmuenzenmeyer) [inline apps/site/i18n/config.json]:
```suggestion
```

**Review by bmuenzenmeyer — COMMENTED** [review]
now we can remove this, since crowdin added `ur` as a custom language

**Pull request merged**